src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -1898,6 +1899,9 @@ private static TimeZoneInfoResult TryGetTimeZone(string id, bool dstDisabled, ou
                             }
                         }
 
+                        cachedData._timeZonesUsingAlternativeIds ??= new Dictionary<string, TimeZoneInfo>(StringComparer.OrdinalIgnoreCase);
+                        cachedData._timeZonesUsingAlternativeIds[id] = zone;
```

Comment:
This could be used from multiple threads concurrently?  If so, this isn't thread-safe and could end up corrupting the dictionary.

Reply:
@stephentoub this method is already called inside the lock on the cachedData object. 

https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs,1856
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs,303
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs,771